### PR TITLE
libsvg: fix for ARM/M1 Macs

### PR DESCRIPTION
**Pull request by citelao**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

I introduced a small patch to the very old libsvg automake configuration file that unblocks building. The patch was based on [a change](https://cgit.freedesktop.org/cairo/commit/?id=afdf3917ee86a7d8ae17f556db96478682674a76) Cairo (the biggest known consumer of libsvg, by chance) made to their own configuration in 2008. It is unlikely that this change will be accepted upstream, since I believe the official source is dead and it's largely hosted by Cairo.

TODO: fix test & style errors as necessary.

### CODE REVIEW

File: Formula/libsvg.rb
Comment (carlocab) [inline Formula/libsvg.rb:12]:
```suggestion
  # Allow building on M1 Macs.
  patch :DATA
```

**Review by carlocab — COMMENTED** [review]
Thanks, @citelao.

Please rebase your branch against the master branch (with `git rebase`, not `git merge`) to get rid of your merge commit. (@BrewTestBot doesn't like them, hence the "automerge skip" label)

Oh, and please squash your commits together, and use the preferred commit style for homebrew/core. We put the name of the formula first in commit-message headings. Refer to the [documentation](https://docs.brew.sh/Formula-Cookbook#commit) for more details.

**Comment by carlocab** [comment]
> It is unlikely that this change will be accepted upstream, since I believe the official source is dead and it's largely hosted by Cairo.

Can you try submitting a patch anyway? It might not be accepted, but we should at least try to send it upstream.

**Comment by citelao** [comment]
> > It is unlikely that this change will be accepted upstream, since I believe the official source is dead and it's largely hosted by Cairo.
> 
> Can you try submitting a patch anyway? It might not be accepted, but we should at least try to send it upstream.

Will do! I'm emailing the Cairo graphics mailing list.

File: Formula/libsvg.rb
Comment (carlocab) [inline Formula/libsvg.rb:41]:
```suggestion
  # Allow building on M1 Macs. This patch is adapted from
  # https://cgit.freedesktop.org/cairo/commit/?id=afdf3917ee86a7d8ae17f556db96478682674a76
```
Let's cut this comment down a little.

When you make this change, please squash your commits together and use the message
```
libsvg: fix for ARM/M1 Macs
```
Feel free to tweak the wording as long as you start the message with `libsvg`.

Thanks again.

**Comment by citelao** [comment]
Unfortunately, I have not received any updates from the Cairo team. I will probably reach out to them again and in the meantime continue iterating on this PR (and handling the feedback).

It is currently stalled since my rebase unveiled some other build errors with newer versions of Automake (can't remember what exactly).

@carlocab informational question—doesn't the PR bot auto-squash for me? I plan to squash my commits anyway, but I'm surprised about this choice. I'm used to committing frequently & then squashing on merge.

**Comment by carlocab** [comment]
BrewTestBot does squash commits together automatically, but then it uses the commit message heading `libsvg: revision`, which is less informative than I'd like. Or a maintainer might trigger a merge manually, but forget to pass `--autosquash`, so the commits aren't squashed.

This PR should've built using the newest version of `automake` with no errors, so I'm not sure what errors you discovered.

**Comment by citelao** [comment]
> BrewTestBot does squash commits together automatically, but then it uses the commit message heading `libsvg: revision`, which is less informative than I'd like. Or a maintainer might trigger a merge manually, but forget to pass `--autosquash`, so the commits aren't squashed.

Makes sense.

> This PR should've built using the newest version of `automake` with no errors, so I'm not sure what errors you discovered.

Yeah, I have yet to look into it deeply. Could be a silly thing.

**Comment by areese** [comment]
This seems to be working with the latest automaker.
The only issue I'm seeing is  ```brew audit --strict libsvg``` fails due to not having any tests.

**Comment by Bo98** [comment]
Happy to merge this once the comment suggestion is addressed and squashed.

(I'm happy to do this for you if you select the checkbox to.)

**Comment by areese** [comment]
I added tests for libsvg in this pr for what it's worth: https://github.com/Homebrew/homebrew-core/pull/75985

**Review by Bo98 — APPROVED** [review]

**Comment by citelao** [comment]
Thank you for working through this! I'll validate the change soon.

**Comment by citelao** [comment]
I can confirm that `brew install --build-from-source libsvg` with the latest changes (including this patch and the new tests) works on my M1 MacBook. Thank you for driving this to completion!

**Comment by areese** [comment]
@citelao This also enables libsvg-cairo: https://github.com/Homebrew/homebrew-core/pull/75957 and once that is built, svg2png and svg2pdf will work.

**Pull request merged**
